### ISSUE TITLE
🐛 BUG: Markdown Component footnotes with custom components

### ISSUE THREAD

**Issue by K3TH3R** [snowpackjs/astro#1309]
### What package manager are you using?

yarn

### What operating system are you using?

Mac

### Describe the Bug

Essentially, I have an Astro template using the Markdown component like this with footnotes:
```jsx
<DocsPage title={title} description={description}>
  <Markdown>
    Dis ligula ullamcorper ante accumsan eget nunc dictum vehicula natoque libero sit duis, enim tristique suspendisse fringilla fusce imperdiet rhoncus venenatis etiam dignissim sodales.[^1]
    [^1]: Here is a new footnote with [a link](https://www.google.com)

    Ipsum ut vel fermentum congue vivamus morbi risus, curabitur consectetur eu interdum eros integer, velit dui curae ligula amet turpis.
    <HeaderLink client:visible>### Example Title</HeaderLink>
    Netus tempus aptent consequat feugiat finibus suscipit molestie sem ridiculus, class commodo lobortis metus fusce eleifend mauris mollis, etiam id urna vivamus fringilla dis ipsum lacinia. Sodales egestas elementum ac commodo curae est morbi quisque maximus, sit venenatis dui lacinia vulputate quis lacus ipsum inceptos vitae, in aliquam aptent nulla laoreet consectetur mauris eros.
  </Markdown>
</DocsPage>
```
`<HeaderLink>` is a Vue component.

I would expect the footnotes to be placed at the bottom of the content block based on the examples of how `remark-footnotes` works. However, they're placed at the end of the first block of markdown content (directly above the `<HeaderLink>`) component. If I remove the `<HeaderLink>` component and leave it as just a basic markdown title, the footnotes migrate properly to the bottom of the content block. 

### Steps to Reproduce

1. git clone git@github.com:K3TH3R/astro-footnotes-bug.git
2. npm install & npm start
3. Notice how the footnotes aren't at the bottom of the page, but are above the first instance of the Vue component.

### Link to Minimal Reproducible Example (Optional)

https://github.com/K3TH3R/astro-footnotes-bug

**Comment by jasikpark** [comment]
Ah, so it sounds like the markdown is getting parsed as 3 different documents then? `[Above <Headerlink>, <Headerlink>, Below <Headerlink>]` If my wild conjecture is true, I guess we'd need to join the markdown before rendering.. 

In a way, https://github.com/snowpackjs/astro/issues/495 should fix this by joining the docs differently 🤔

**Comment by K3TH3R** [comment]
Yes, it was my suspicion as well that it was parsed as 3 different documents.

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.
